nix/overlay.nix

```diff
@@ -0,0 +1,111 @@
+      };
+    fetchRustyV8Hashes = writeShellScriptBin "fetch-librusty_v8-hashes.sh" ''
+      version=$(${getExe' yq "tomlq"} '.package | map(select(.name == "v8")) | .[0].version' --raw-output ${inputs.self}/Cargo.lock)
+      echo "Update ./nix/overlay.nix as follows:"
+      echo "RUSTY_V8_ARCHIVE = fetchRustyV8 \"$version\" {"
+      for system in ${concatStringsSep " " config.systems}; do
+          target=$(nix eval --raw nixpkgs#legacyPackages.$system.stdenv.hostPlatform.rust.rustcTarget)
+          hash=$(nix-prefetch-url --print-path ${buildRustyV8Url "$version" "$target"} | tail -n 1 | xargs nix hash file)
+          echo "  $system = \"$hash\";"
+      done
+      echo "};"
+    '';
+    pname = "gauntlet";
+    version = "v11";
+    src = inputs.self;
+    cargoArtifactsArgs = {
+      inherit pname src version;
+      cargoExtraArgs = "--features release";
+      nativeBuildInputs = [cmake pkg-config protobuf];
+      buildInputs = [openssl];
+      # OPENSSL_CONFIG_DIR didn't work for vendored dependencies
+      OPENSSL_NO_VENDOR = true;
+      RUSTY_V8_ARCHIVE = fetchRustyV8 "130.0.2" {
+        aarch64-darwin = "sha256-aWZ/4Q4Wttx37xOdBmTCPGP+eYGhr4CM1UkYq8pC7Qs=";
+        aarch64-linux = "sha256-p9+tHmKIM5wBABubHIAstpwfzO19ypPzOuaV4b6loCU=";
+        x86_64-darwin = "sha256-zNC0DAkMbbFM1M+t6rgKtN0QAm4ONEbCi6Sxivhf8dk=";
+        x86_64-linux = "sha256-ew2WZhdsHfffRQtif076AWAlFohwPo/RbmW/6D3LzkU=";
+      };
+    };
+    craneLib = inputs.crane.mkLib final;
+    cargoArtifacts = craneLib.buildDepsOnly cargoArtifactsArgs;
+  in {
+    gauntlet = craneLib.buildPackage {
+      inherit (cargoArtifactsArgs) cargoExtraArgs OPENSSL_NO_VENDOR RUSTY_V8_ARCHIVE;
+      inherit cargoArtifacts pname src version;
+      # fetchNpmDeps + makeCacheWritable is necessary with npm git:// dependencies
+      npmDeps = fetchNpmDeps {
+        inherit src;
+        name = "${pname}-${version}-npm-deps";
+        hash = "sha256-cehjrUlEVcqe+tOPa5Ba4zUAFvD+C5I5Cd62aM1d0PU=";
+      };
+      makeCacheWritable = true;
+      nativeBuildInputs = cargoArtifactsArgs.nativeBuildInputs ++ [nodejs npmConfigHook] ++ optional isLinux makeWrapper;
+      buildInputs = cargoArtifactsArgs.buildInputs ++ [deno];
+      preBuild = "npm run build";
+      postInstall =
+        if isLinux
+        then ''
+          install -Dm644 assets/linux/gauntlet.desktop $out/share/applications/gauntlet.desktop
+          install -Dm644 assets/linux/gauntlet.service $out/lib/systemd/user/gauntlet.service
+          install -Dm644 assets/linux/icon_256.png $out/share/icons/hicolor/256x256/apps/gauntlet.png
+        ''
+        else ''
+          contentsDir=$out/Applications/Gauntlet.app/Contents
+          install -Dm755 $out/bin/gauntlet $contentsDir/MacOS/Gauntlet
+          install -Dm644 assets/macos/AppIcon.icns $contentsDir/Resources/AppIcon.icns
+          install -Dm644 assets/macos/Info.plist $contentsDir/Info.plist
```

Comment:
Currently Info.plist contains `__VERSION__` version placeholder which is replaced when packaging, this should probably be handled as well?

Reply:
Nice catch, just fixed this